CONTRIBUTING.md

```diff
@@ -0,0 +1,79 @@
+
+**Board:** (e.g Arduino Uno, Intel Edison, etc)
+**Shield:** (if you're using one, what kind)
+**Hardware you are having an issue with:** (e.g. servos, leds, sensors, and what brand/type you are using)
+**Version of Johnny-Five:**
+**What your expectations are:**
+**What the actual outcome is:**
+**Steps to reproduce (including code samples):**
+
+Lastly, if you are able, including a video is incredibly helpful in a lot of cases of debugging issues. You can upload videos to [youtube](https://www.youtube.com/), [vine](https://vine.co/), or any site that lets you host videos.
+
+If the issue has been reported before but you have new information to help troubleshoot the issue, add a comment to the thread with the same information as requested above.
+
+
+<a name="requesting-features"></a>
+## Requesting Features
+
+To request a new feature be added to one of the existing classes, create a [github issue](https://github.com/rwaldron/johnny-five/issues) and include:
+
+**What feature you'd like to see:**
+**Why this is important to you:** (this is here because it's interesting knowing what cool things people are working on and also could help community members make suggestions for work-arounds until the feature is built)
+
+
+
+<a name="hardware-support"></a>
+## Hardware Support
+
+Often, people get a new cool toy and wonder "does johnny five support this yet?"  When requesting support for new hardware remember that the team working on johnny-five might not own that hardware.
+
+To submit a request to support new hardware, first create a [github issue](https://github.com/rwaldron/johnny-five/issues) and include a link to the specs of the hardware and where one can purchase it. Often, it might be suggested that you be the one to help impliment the support yourself if you're already in possession of the product.
+
+<a name="pullrequests"></a>
+## Submitting Pullrequests
+
+To contribute code to Johnny Five, fork the project onto your github account and do your work in a branch. Before you submit the PR, make sure to rebase master into your branch so that you have the most recent changes and nothing breaks or conflicts.  Lint and test your code using [grunt](https://github.com/gruntjs/grunt). Also squash your commits to a reasonable size before submitting.
+
+If you are contributing code, it must include unit tests that fail if the working code isn't present and succeed when it is. Also make sure you run `grunt jsbeautifier` to fix any syntax issues.
+
+All contributions must adhere to the [Idiomatic.js Style Guide](https://github.com/rwldrn/idiomatic.js),
+by maintaining the existing coding style.
+
+
+<a name="writing-tests"></a>
+## Writing Tests
+
+Test are written using [sinon](http://sinonjs.org/).  If you are having issues making a test pass, ask for help in the Johnny-Five [gitter](https://gitter.im/) room.  Tests can be the hardest part to write when contributing code, so don't be discouraged.
+
```

Comment:
Test are written using [nodeunit[(https://github.com/caolan/nodeunit) and [sinon](http://sinonjs.org/).